server/consumer.go

```diff
@@ -1532,24 +1532,34 @@ func (o *consumer) updateConfig(cfg *ConsumerConfig) error {
 // of the "forceStart" parameter. It then returns what the final value
 // for InactiveThreshold will be (so that the jittery default for ephemeral
 // consumers can be captured in config).
-func (o *consumer) updateInactiveThresholdLocked(threshold time.Duration, forceStart bool) time.Duration {
-	started := o.dtmr != nil
+func (o *consumer) updateInactiveThresholdLocked(threshold time.Duration) time.Duration {
+	origThresh := o.dthresh
 	defer func() {
-		if forceStart || started {
-			stopAndClearTimer(&o.dtmr)
+		if origThresh == o.dthresh {
+			// Nothing has changed, so don't do anything.
+			return
+		}
+		// Otherwise, reset the timer, and if the new threshold is greater
+		// than zero, start it again from this point.
+		stopAndClearTimer(&o.dtmr)
+		if o.dthresh > 0 {
 			o.dtmr = time.AfterFunc(o.dthresh, func() { o.deleteNotActive() })
 		}
 	}()
-	if threshold > 0 {
+	if o.isDurable() || threshold > 0 {
+		// Allow setting a durable consumer threshold to either a specific
+		// value or to 0 (effectively disabling it). Also allow setting an
+		// ephemeral consumer to a non-zero value.
 		o.dthresh = threshold
```

Comment:
I was not asking to make the changes yet, since I wanted @derekcollison input to see if this is something we want to support (reset durable to 0).

If we do and you want to keep this code, you now have for the isDurable() case, make sure that threshold has not been given as a negative value. So anything < 0 should become 0 I guess.

Reply:
I'll revert for now but leave the commits on the branch rather than rebasing.

Reply:
It sounds like might be reasonable to set back to 0, although I want in general for systems to avoid that since we have issues with abandoned durables taking up resources in systems.